ui/src/views/Clients/ViewClient/index.jsx

```diff
@@ -84,20 +86,22 @@ export default class ViewClient extends Component {
   };
 
   handleDeleteClient = async clientId => {
-    this.setState({ error: null, loading: true });
+    this.setState({ dialogError: null, loading: true });
 
-    try {
-      await this.props.client.mutate({
-        mutation: deleteClientQuery,
-        variables: { clientId },
-      });
+    await this.props.client.mutate({
+      mutation: deleteClientQuery,
+      variables: { clientId },
+    });
 
-      this.setState({ error: null, loading: false });
+    this.props.history.push(`/auth/roles`);
```

Comment:
This line should be done inside `handleDialogActionComplete` since it has nothing to do with the asynchronous action.